### DISCUSSION TITLE
Casting strings removes the length parameter?

### DISCUSSION THREAD

**Discussion by kesmit13** [ibis-project/ibis#4578] [Q&A] [answered]
I've been working with some casting issues while writing a backend for SingleStoreDB. The SingleStoreDB is based on MySQL so it uses the same drivers, but there are SQL differences. I was doing some work with casts and noticed that you couldn't cast doubles, which is true of older versions of MySQL as well. SingleStoreDB adds a casting operation (`:>`) that can cast doubles, but also casts strings differently than the `cast` function. If you don't specify a length, it can truncate numbers. For example, doing `cast x :> string` where x is a double value 1000 will return '1'. If you use `cast x :> char(10)` then you get the full value as expected. However, Ibis appears to throw away the length value when doing casts. I wondered if anyone knew the reasoning why?

**Comment by cpcloud** [comment, marked as answer]
Thanks for opening an issue.

This behavior is ... interesting.

Ibis throws away the length parameter, because up until this discussion there are no string APIs (or generic APIs like `.cast` that use strings) where the length matters. All of our backends handle variable length strings so it didn't seem worth it to have another string type.

I don't think this behavior is enough to justify adding a whole new type, but you can special case the `ops.Cast` translation and look at the operation's `output_dtype` attribute as well as the input dtypes and choose your desired cast code.

Would that work for you?

**Comment by kesmit13** [comment]
I'll have to investigate further, but if the `output_dtype` contains that information, that should be fine. I already override the cast function in our backend.
